include/utils/filo.h

```diff
@@ -0,0 +1,120 @@
+#ifndef _UTILS_FILO_H_
+#define _UTILS_FILO_H_
+
+/*
+ * Copyright (c) 2020 Abhishek Arora <abhiarora@gotomain.io>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct {
+    void *buffer;
+    unsigned top;
+    unsigned elem_size;
+    unsigned max_size;
+} filo_t;
+
+#define FILO_DEF(type, name, sz) \
+    type name_##buffer[sz]; \
+    filo_t name = { .buffer = name ## _buffer, \
+                     .top = 0, \
+                     .elem_size = sizeof(type),\
+                     .max_size = sz \
+                    }; \
+
+#define FILO_GLOB_DEF(type, name, sz) \
+    FILO_DEF(type, name, sz) \
+    int name ## _push(const type *elem) \
+    { \
+        return filo_push(&name, (void *)elem); \
+    } \
+    int name ## _pop(const type *elem) \
+    { \
+        return filo_pop(&name, (void *)elem, true); \
+    } \
+    int name ## _peek(const type *elem) \
+    { \
+        return filo_pop(&name, (void *)elem, false); \
+    } \
+    unsigned name ## _get_count(const type *elem) \
+    { \
+        return filo_get_count(&name); \
+    } \
+    int name ## _reset(const type *elem) \
+    { \
+        return filo_reset(&name); \
+    } \
+
```

Comment:
You must provide macro helpers such as `FILO_PUSH(name, element)` to call these methods. The exact definition of these type wrapper methods must be kept opaque to consumers.